### ISSUE TITLE
CoAP over TCP support

### ISSUE THREAD

**Issue by luochen52** [eclipse-wakaama/wakaama#448]

**Comment by luochen52** [comment]
and is lw2m2 over TCP supported ? if supported, how can I enable it?

**Comment by JRubis** [comment]
Yes, that is a good question and I was wondering it myself. It seems that @sbertin-telular has done a lot of work on v1.1, but his changes have not been merged.

**Comment by sbertin-telular** [comment]
My changes are all available at https://github.com/sbertin-telular/wakaama/tree/lwm2m_1.1. Not everything has had pull requests created due to dependencies on other open pull requests (it gets messy). TCP support is not yet available.

There are still some items not yet implemented (no particular order):
•	New attributes (Minimum/Maximum evaluation period) and attributes at the resource instance level. Attributes as a whole could use a rework to be compliant.
•	Write-Composite. With the current Wakaama design I’m not sure this can be implemented atomically.
•	Observe-Composite
•	Cancel Observation-Composite
•	TCP binding. @dnav  has said (wakaama-dev 2/20/2019) that he has already implemented it, but it hasn’t been released yet.
•	LWM2M 1.1.1 compatibility changes
•	New optional resources in objects for the client example
•	OSCORE. @dnav  has said (wakaama-dev 2/20/2019) someone is working on this.
•	Enforcing access control. LWM2M 1.1.1 makes this optional for an “Access Control-Disabled” configuration where only 1 server is supported.

**Comment by JRubis** [comment]
@sbertin-telular I was trying to follow the mail list discussion between you and @dnav concerning v1.1 planning. It wasn't clear to me in the end if the re-design of Wakaama he mentioned was or wasn't in scope for v1.1?

**Comment by sbertin-telular** [comment]
I don't know how extensive the redesign he had in mind is, but he started it with #431. My lwm2m_1.1 branch does incorporate that.

**Comment by sbernard31** [comment]
There is some new on Wakaama (see #488)

@sbertin-telular do you have any plan for this redesign ? 

Should we close this issue ?

**Comment by sbertin-telular** [comment]
Beyond the code reorganization in #431 I don't have plans for the redesign.

This issue started with LWM2M over TCP, which is something I haven't addressed. I would suggest creating an issue whose title reflects that and close this issue.

**Comment by sbernard31** [comment]
> This issue started with LWM2M over TCP, which is something I haven't addressed. I would suggest creating an issue whose title reflects that and close this issue.

Just to let you know, "Issue title" can be changed (`Edit` button near the issue title).
Both solution (rename /close&create) sounds fine too. Please do what you prefer :pray:

**Comment by sbertin-telular** [comment]
> Just to let you know, "Issue title" can be changed (`Edit` button near the issue title).

You may be able to change the title, but I don't have an Edit button.

> Both solution (rename /close&create) sounds fine too. Please do what you prefer 🙏

I also can't close the issue. I suggest renaming it to "CoAP over TCP support" and changing the label to enhancement. That will keep the record of who is interested in the change.

**Comment by LukasWoodtli** [comment]
Closing because the Wakaama project will be terminated: See issue #883 and [iot.wakaama 2026.05 Termination Review](https://gitlab.eclipse.org/eclipsefdn/emo-team/emo/-/issues/1191)
